pkgs/http/test/html/cache_test.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+@TestOn('browser')
+library;
+
+import 'dart:async';
+
+import 'package:http/browser_client.dart';
+import 'package:http/http.dart' as http;
+import 'package:http/http.dart';
+import 'package:test/test.dart';
+
+import 'utils.dart';
+
+void main() {
+  late Uri url;
+  setUp(() async {
+    final channel = spawnHybridUri(Uri(path: '/test/stub_server.dart'));
+    var port = (await channel.stream.first as num).toInt();
+    url = echoUrl.replace(port: port);
+  });
+
+  test('#send a GET with default type', () async {
+    var client = BrowserClient(cacheMode: CacheMode.defaultType);
```

Comment:
You are only doing a single request here so I'm not sure how this is testing the caching behavior.

Reply:
Nope, it is two GET requests, when it is set to default type it is cached twice.

Reply:
I guess, I need to drain the request so, I can see the difference.